### PR TITLE
Remote wallet

### PR DESCRIPTION
**Pull request by alonmuroch**
A PR for remote wallet key manager

TODO(you): choose "part of" or "resolves" and the associated github issue #.

[Part of|Resolves] #531 

---

# Description

**Write why you are making the changes in this pull request**

**Write a summary of the changes you are making**

**Link anything that would be helpful or relevant to the reviewers**

### CODE REVIEW

File: validator/client/validator_aggregate.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package client
 import (
 	"context"
 	"fmt"
+	"github.com/prysmaticlabs/prysm/shared/bls"
```
Comment (terencechain) [inline validator/client/validator_aggregate.go]: goimports

File: validator/keymanager/remote_wallet.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+	r, err := km.rw.SignAttestation(ctx, &RW.SignAttestationRequest{PublicKey:pubKey[:], Data:data, Domain:domain})
+	if err != nil {
+		return nil, err
+	}
+
+	return bls.SignatureFromBytes(r.Sig)
+}
+
+// SignSlot signs an aggregate attestation for the validator to broadcast.
```
Comment (terencechain) [inline validator/keymanager/remote_wallet.go]:
```suggestion
// SignSlot signs the selection proof of an aggregated attestation for the validator to broadcast.
```

File: validator/keymanager/remote_wallet.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	if err != nil {
+		log.Fatalf("did not connect to remote wallet: %v", err)
+	} else {
+		log.Printf("connected to remote wallet at: %v", opts.Url)
+	}
+	//defer conn.Close() TODO
+	c := RW.NewRemoteWalletClient(conn)
+
+	km := &RemoteWallet{
+		keys: make([][48]byte,0),
```
Comment (terencechain) [inline validator/keymanager/remote_wallet.go]: can we define `48` as a constant somewhere?

Comment (alonmuroch) [reply validator/keymanager/remote_wallet.go]: I was thinking that as well. It makes a lot of sense. 
I'll try and do a different PR for that

File: validator/client/validator_aggregate.go (hunk 1 above)
Comment (alonmuroch) [reply validator/client/validator_aggregate.go]: what do you mean?

Comment (prestonvanloon) [reply validator/client/validator_aggregate.go]: Please run `goimports -w validator/client/validator_aggregate.go` these imports are not sorted properly.

File: validator/keymanager/remote_wallet.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	grpc "google.golang.org/grpc"
+	"time"
+)
+
+type RemotewalletOpts struct {
+	Url    string   `json:"url"`
+}
+
+// TODO
```
Comment (prestonvanloon) [inline validator/keymanager/remote_wallet.go]: What is the TODO here?

File: WORKSPACE [hunk 5]
```diff
@@ -1583,6 +1583,13 @@ go_repository(
     importpath = "github.com/wealdtech/go-indexer",
 )
 
+local_repository(
```
Comment (prestonvanloon) [inline WORKSPACE]: Please update

**Comment by prestonvanloon** [comment]
This this PR relevant #5133

**Comment by alonmuroch** [comment]
@prestonvanloon I saw you merged @mcdee PR so I guess this one is irrelevant.

**Comment by prestonvanloon** [comment]
Alright, thanks @alonmuroch! Closing this in favor of #5133

**Pull request closed without merging**